dbms/src/Functions/FunctionsHashing.h

```diff
@@ -956,6 +956,136 @@ class FunctionURLHash : public IFunction
     }
 };
 
+struct JavaHashImpl
+{
+    static Int32 apply(const char * data, const size_t size)
+    {
+        Int32 h = 0;
+        for (int i = 0; i < (int)size; ++i)
+        {
+            h = 31 * h + data[i];
+        }
+        return h;
+    }
+};
```

Comment:
Is it real, that Java still uses such a poor hash function?

Reply:
It is a common fast implementation for "==" operator in Java. If two objects has the same hashcode, then method object1.equals(object2) is called. E.g. this is a fast part of contract, that allows not to perform deep comparison of two complicated objects.

Reply:
@alexey-milovidov yes, it's poor, but used wildly in java world